test/with_dummyserver/test_proxy_poolmanager.py

```diff
@@ -571,19 +619,13 @@ def test_https_proxy_tls_error(
 
     @requires_network()
     @pytest.mark.parametrize(
-        ["proxy_scheme", "use_forwarding_for_https"],
+        ["proxy_scheme"],
         [
-            ("http", False),
-            ("https", False),
-            ("https", True),
+            ("http",),
+            ("https",),
         ],
     )
-    def test_proxy_https_target_tls_error(
```

Comment:
Can you expand on why you disabled this test?

> I enabled the failing test, but because the test doesn't really make sense (I think) for when a forwarding proxy is used, I removed that case from the test. 

Shouldn't we verify that setting ssl_context will raise in the ``https`` proxy scheme case and that the context is not expected on ``http`` proxy scheme case.

Reply:
My thought was that instead of adding a condition in the test itself for (https proxy, use forwarding), I instead created ```test_forwarding_for_https_error_with_ssl_ctx``` and ```test_forwarding_for_https_error_with_only_ssl_ctx``` to test for ssl_context being set, as the scenario is unique for these proxy settings. However, I can add it back and handle the expected exception in the test itself if that would be better.

For the http proxy case, currently passing a ```proxy_ssl_context``` does not result in any behavior change or warning. Should a warning be added, recommending that ssl_context be set instead?

Reply:
Reviewing this more carefully, the changes you did are correct. I don't see much value in keeping the ``use_forwarding_for_https`` flag if it's being skipped. 

> For the http proxy case, currently passing a proxy_ssl_context does not result in any behavior change or warning. Should a warning be added, recommending that ssl_context be set instead?

That's expected IMO and raising a warning might be confusing. Happy to discuss this more.